ads/nativo.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS-IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript} from '../3p/3p';
+
+export function nativo(global, data) {
+
+  let ntvAd;
+  (function(ntvAd, global, data) {
+    global
+      .frames
+      .location
+      .hash = global.frames.location.hash.replace(/({).*(})/,'');
+
+    ///
+    // Private
+    ///
+    let delayedAdLoad = false;
+    let percentageOfadViewed;
+    const loc = global.context.location;
+
+    function isValidDelayTime(delay) {
+      return ((typeof delay != 'undefined'
+        && !isNaN(delay)
+        && parseInt(delay,10) >= 0));
+    }
+    function isDelayedTimeStart(data) {
+      return (isValidDelayTime(data.delayByTime)
+        && ('delay' in data)
+        && !('delayByView' in data));
+    }
+    function isDelayedViewStart(data) {
+      return (isValidDelayTime(data.delayByTime)
+        && ('delayByView' in data));
+    }
+    function loadAdWhenViewed() {
+      const g = global;
+      global.context.observeIntersection(function(positions) {
+        const coordinates = getLastPositionCoordinates(positions);
+        if ((coordinates
+          .intersectionRect
```

Comment:
This is unfortunately not currently covered by our lint, but practice in the repo.